### PR TITLE
Webhook route & a lot of changes 

### PR DESCRIPTION
**Pull request by NLDev**
Added and fixed a lot of stuff:

- Added github webhook 
- Added command execution for webhook
- Fixed Gallery error: "Can't set headers after they are sent."
- Generally refactored gallery controller
- Moved routes from app.ts to /controller/router.ts
- Strip language query string from URL 
- Added Webhook things to config 
- Added a title to the header
- Added auto-generated titles for subsites / subroutes 
- Fixed language handler (read cookie data)

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-        cookieprefix: "_sti",
-        localhost_only: false
+    "server": {
+        "port": 3000,
+        "cookieprefix": "_sti",
+        "localhost_only": false,
+        "hook": {
+            "githook_secret": "",
+            "githook_commands": [
+                "git pull"
```
Comment (kellerrennkadse) [inline src/config.ts]: is it really enough to just pull from github? Don't we need to recompile some stuff?

Comment (NLDev) [reply src/config.ts]: That's why it's inside the config. It can be customized

Comment (kellerrennkadse) [reply src/config.ts]: the `npm start` command will most likely crash, as there's already another process listening on port 3000. We would need to **re**start the server

Comment (NLDev) [reply src/config.ts]: Yes, we have talked about this in the dev chat yesterday. As mentioned, the commands are meant to be adjusted on the prod server. I have talked to @Insax and suggested the use of [pm2](http://pm2.keymetrics.io/).

**Pull request merged**